x-pack/plugins/infra/public/components/logging/log_text_stream/loading_item_view.tsx

```diff
@@ -6,144 +6,272 @@
 
 /* eslint-disable max-classes-per-file */
 
-import { EuiButtonEmpty, EuiIcon, EuiProgress, EuiText } from '@elastic/eui';
-import { FormattedMessage, FormattedRelative } from '@kbn/i18n/react';
+import {
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiLoadingSpinner,
+  EuiButton,
+} from '@elastic/eui';
+import { FormattedMessage, FormattedTime, FormattedRelative } from '@kbn/i18n/react';
 import * as React from 'react';
+import { Unit } from '@elastic/datemath';
 
 import { euiStyled } from '../../../../../observability/public';
+import { LogTextSeparator } from './log_text_separator';
+import { extendDatemath } from '../../../utils/datemath';
+
+type Position = 'start' | 'end';
 
 interface LogTextStreamLoadingItemViewProps {
-  alignment: 'top' | 'bottom';
+  position: Position;
+  /** topCursor.time || bottomCursor.time */
```

Comment:
Are these comments left-overs?

Reply:
They try to document the value below. Maybe it's clearer if I put them in the same line?

Reply:
That might be helpful. And maybe a few words that put them into context such as

```typescript
  timestamp?: number; // either the top or the bottom cursor's timestamp
```

?